src/edu/csus/ecs/pc2/clics/API202306/CLICSContestAccess.java

```diff
@@ -0,0 +1,167 @@
+    public static final String API_CAPABILITY_TEAM_CLAR = "team_clar";
+    public static final String API_CAPABILITY_PROXY_SUBMIT = "proxy_submit";
+    public static final String API_CAPABILITY_PROXY_CLAR = "proxy_clar";
+    public static final String API_CAPABILITY_ADMIN_SUBMIT = "admin_submit";
+    public static final String API_CAPABILITY_ADMIN_CLAR = "admin_clar";
+    // tentative - not official, but mentioned in the spec
+    public static final String API_CAPABILITY_AWARDS_UPDATE = "awards_update";
+
+    public static final String ENDPOINT_ACCESS_METHOD = "getEndpointProperties";
+
+    @JsonProperty
+    private String [] capabilities;
+
+    @JsonProperty
+    private CLICSEndpoint [] endpoints;
+
+    IInternalController controller = null;
+
+    /**
+     * Fill in properties for contest state as per 2023-06 spec
+     *
+     * @param sc security info for user making request
+     * @param model
+     * @param controller
+     * @param contestId
+     */
+    public CLICSContestAccess(SecurityContext sc, IInternalContest model, IInternalController controller, String contestId) {
+        this.controller = controller;
+
+        // For each role the connected user has, we enumerate what they can do with each endpoint.
+        ArrayList<String> cap = new ArrayList<String>();
+
+        if(ContestService.isContestStartAllowed(sc)) {
+            cap.add(API_CAPABILITY_CONTEST_START);
+        }
+        if(ContestService.isContestThawAllowed(sc)) {
+            cap.add(API_CAPABILITY_CONTEST_THAW);
+        }
+        if(SubmissionService.isTeamSubmitAllowed(sc)) {
+            cap.add(API_CAPABILITY_TEAM_SUBMIT);
+        }
+        if(ClarificationService.isTeamSubmitClarificationAllowed(sc)) {
+            cap.add(API_CAPABILITY_TEAM_CLAR);
+        }
+        if(SubmissionService.isProxySubmitAllowed(sc)) {
+            cap.add(API_CAPABILITY_PROXY_SUBMIT);
+        }
+        if(ClarificationService.isProxySubmitClarificationAllowed(sc)) {
+            cap.add(API_CAPABILITY_PROXY_CLAR);
+        }
+        if(SubmissionService.isAdminSubmitAllowed(sc)) {
+            cap.add(API_CAPABILITY_ADMIN_SUBMIT);
+        }
+        if(ClarificationService.isAdminSubmitClarificationAllowed(sc)) {
+            cap.add(API_CAPABILITY_ADMIN_CLAR);
+        }
+        if(AwardService.isAwardsUpdateAllowed(sc)) {
+            cap.add(API_CAPABILITY_AWARDS_UPDATE);
+        }
+        capabilities = cap.toArray(new String[0]);
+
+        /*
+         * As per the spec, here are the possible endpoints.
+         * These should match the class names in this package, eg. Contest -> ContestService, Run -> RunService, etc.
+         */
+        String [] serviceNames = { "Contest", "JudgementType", "Language", "Problem", "Group", "Organization",
+                "Team", "Person", "Account", "State", "Submission", "Judgement", "Run", "Clarification",
+                "Scoreboard", "EventFeed", "Award", "Commentary"
```

Comment:
There are two endpoints missing from this list of service names:  `API Version` (i.e., `/`), and `Webhooks`.  I'm assuming the latter is "SOMEDAY"... but isn't "API Version" something we need to implement now?